### PR TITLE
Convert imports in generate_imports into one per line, so that it can…

### LINKED ISSUE #13036: Convert imports in generate_imports into one per line
Right now there's a long list in `CodegenRust.py` like:
```python
        'js::jsapi::{Handle, HandleId, HandleObject, HandleValue, HandleValueArray}',
        'js::jsapi::{INTERNED_STRING_TO_JSID, IsCallable, JS_AtomizeAndPinString}',
```
These are in alphabetical order, which is great. However, adding new imports to this list is a pain, because we need to re-wrap all the subsequent lines that become too long. We should convert it to a list of one import per line, so it's easy to add new ones and they remain in alphabetical order.

### PR DESCRIPTION
**Pull request by tp6vup54**
Convert imports in generate_imports into one per line, so that it can easily add new ones with alphabetical order.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #13036 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

… easily add new ones with alphabetical order.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13070)

### CODE REVIEW

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 1]
```diff
@@ -2166,11 +2166,14 @@ def UnionTypes(descriptors, dictionaries, callbacks, config):
...
         'dom::bindings::js::Root',
-        'dom::bindings::str::{ByteString, DOMString, USVString}',
+        'dom::bindings::str::ByteString',
+        'dom::bindings::str::DOMString',
+        'dom::bindings::str::USVString',
         'dom::types::*',
         'js::error::throw_type_error',
         'js::jsapi::JSContext',
-        'js::jsapi::{HandleValue, MutableHandleValue}',
+        'js::jsapi::HandleValue',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Looks like this line should be above the previous line.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 2]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
 
     return CGImports(cgthings, descriptors, callbacks, dictionaries, enums, [
         'js',
-        'js::{JS_CALLEE, JSCLASS_GLOBAL_SLOT_COUNT}',
-        'js::{JSCLASS_IS_DOMJSCLASS, JSCLASS_IS_GLOBAL, JSCLASS_RESERVED_SLOTS_MASK}',
+        'js::JS_CALLEE',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: `_` is greater than `C`, so this should be put after the `js::JSCLASS_*` declarations.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 3]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'js::jsapi::JS_HasProperty',
+        'js::jsapi::JS_HasPropertyById',
+        'js::jsapi::JS_InitializePropertiesFromCompatibleNativeObject',
+        'js::jsapi::JS_NewObject',
+        'js::jsapi::JS_NewObjectWithGivenProto',
+        'js::jsapi::JS_NewObjectWithoutMetadata',
+        'js::jsapi::JS_SetImmutablePrototype',
+        'js::jsapi::JS_SetProperty',
+        'js::jsapi::JS_SetReservedSlot',
+        'js::jsapi::JS_SplicePrototype',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Same problem as above - `_` is greater than any Latin capital letters.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 4]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
-        'script_runtime::{store_panic_result, maybe_take_panic_result}',
+        'script_runtime::store_panic_result',
+        'script_runtime::maybe_take_panic_result',
         'std::borrow::ToOwned',
         'std::cmp',
         'std::mem',
         'std::num',
         'std::os',
-        'std::panic::{self, AssertUnwindSafe}',
+        'std::panic::{self}',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Remove braces here.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 5]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
         'js::jsval::JSVal',
-        'js::jsval::{ObjectValue, ObjectOrNullValue, PrivateValue}',
-        'js::jsval::{NullValue, UndefinedValue}',
-        'js::glue::{CallJitMethodOp, CallJitGetterOp, CallJitSetterOp, CreateProxyHandler}',
-        'js::glue::{GetProxyPrivate, NewProxyObject, ProxyTraps}',
-        'js::glue::{RUST_JSID_IS_STRING, int_to_jsid, RUST_SYMBOL_TO_JSID}',
+        'js::jsval::ObjectValue',
+        'js::jsval::ObjectOrNullValue',
+        'js::jsval::PrivateValue',
+        'js::jsval::NullValue',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: `N` comes before `O`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 6]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'js::jsval::UndefinedValue',
+        'js::glue::CallJitMethodOp',
+        'js::glue::CallJitGetterOp',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: `G` comes before `M`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 7]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'js::glue::CallJitSetterOp',
+        'js::glue::CreateProxyHandler',
+        'js::glue::GetProxyPrivate',
+        'js::glue::NewProxyObject',
+        'js::glue::ProxyTraps',
+        'js::glue::RUST_JSID_IS_STRING',
+        'js::glue::int_to_jsid',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Lowercase letters after uppercase.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 8]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'dom::bindings::interface::create_global_object',
+        'dom::bindings::interface::create_interface_prototype_object',
+        'dom::bindings::interface::create_named_constructors',
+        'dom::bindings::interface::create_noncallback_interface_object',
+        'dom::bindings::interface::define_guarded_constants',
+        'dom::bindings::interface::define_guarded_methods',
+        'dom::bindings::interface::define_guarded_properties',
+        'dom::bindings::interface::is_exposed_in',
+        'dom::bindings::interface::ConstantVal::IntVal',
+        'dom::bindings::interface::ConstantVal::UintVal',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Put before `is_exposed_in`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 9]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'dom::bindings::iterable::IteratorType',
+        'dom::bindings::iterable::Iterable',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Put above `IteratorType`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 10]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'dom::bindings::js::JS',
+        'dom::bindings::js::Root',
+        'dom::bindings::js::RootedReference',
+        'dom::bindings::js::OptionalRootedReference',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Put above `Root`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 11]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'dom::bindings::utils::is_platform_object',
+        'dom::bindings::utils::resolve_global',
+        'dom::bindings::utils::set_dictionary_property',
+        'dom::bindings::utils::trace_global',
+        'dom::bindings::trace::JSTraceable',
+        'dom::bindings::trace::RootedTraceable',
+        'dom::bindings::callback::CallbackContainer',
+        'dom::bindings::callback::CallbackInterface',
+        'dom::bindings::callback::CallbackFunction',
+        'dom::bindings::callback::CallSetup',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Put above `CallbackContainer`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 12]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
-        'dom::bindings::conversions::{ToJSValConvertible, jsid_to_str, native_from_handlevalue}',
-        'dom::bindings::conversions::{native_from_object, private_from_object, root_from_object}',
-        'dom::bindings::conversions::{root_from_handleobject, root_from_handlevalue}',
-        'dom::bindings::codegen::{PrototypeList, RegisterBindings, UnionTypes}',
-        'dom::bindings::error::{Fallible, Error, ErrorResult}',
+        'dom::bindings::conversions::ConversionBehavior',
+        'dom::bindings::conversions::ConversionResult',
+        'dom::bindings::conversions::DOM_OBJECT_SLOT',
+        'dom::bindings::conversions::IDLInterface',
+        'dom::bindings::conversions::is_array_like',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Put above `jsid_to_str`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 13]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
-        'dom::bindings::conversions::{FromJSValConvertible, StringificationBehavior}',
-        'dom::bindings::conversions::{ToJSValConvertible, jsid_to_str, native_from_handlevalue}',
-        'dom::bindings::conversions::{native_from_object, private_from_object, root_from_object}',
-        'dom::bindings::conversions::{root_from_handleobject, root_from_handlevalue}',
-        'dom::bindings::codegen::{PrototypeList, RegisterBindings, UnionTypes}',
-        'dom::bindings::error::{Fallible, Error, ErrorResult}',
+        'dom::bindings::conversions::ConversionBehavior',
+        'dom::bindings::conversions::ConversionResult',
+        'dom::bindings::conversions::DOM_OBJECT_SLOT',
+        'dom::bindings::conversions::IDLInterface',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Put above `StringificationBehavior`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 14]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'dom::bindings::conversions::is_array_like',
+        'dom::bindings::conversions::FromJSValConvertible',
+        'dom::bindings::conversions::StringificationBehavior',
+        'dom::bindings::conversions::ToJSValConvertible',
+        'dom::bindings::conversions::jsid_to_str',
+        'dom::bindings::conversions::native_from_handlevalue',
+        'dom::bindings::conversions::native_from_object',
+        'dom::bindings::conversions::private_from_object',
+        'dom::bindings::conversions::root_from_object',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Put after `root_from_handlevalue`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 15]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'dom::bindings::conversions::root_from_handleobject',
+        'dom::bindings::conversions::root_from_handlevalue',
+        'dom::bindings::codegen::PrototypeList',
+        'dom::bindings::codegen::RegisterBindings',
+        'dom::bindings::codegen::UnionTypes',
+        'dom::bindings::error::Fallible',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Put after `ErrorResult`.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 16]
```diff
@@ -5321,95 +5324,216 @@ def generate_imports(config, cgthings, descriptors, callbacks=None, dictionaries
...
+        'dom::bindings::str::USVString',
+        'dom::bindings::weakref::DOM_WEAK_SLOT',
+        'dom::bindings::weakref::WeakBox',
+        'dom::bindings::weakref::WeakReferenceable',
         'dom::browsingcontext::BrowsingContext',
         'mem::heap_size_of_raw_self_and_children',
         'libc',
         'util::prefs::PREFS',
-        'script_runtime::{store_panic_result, maybe_take_panic_result}',
+        'script_runtime::store_panic_result',
```
Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Put after `maybe_take_panic_result`.

File: components/script/dom/bindings/codegen/CodegenRust.py (hunk 4 above)
Comment (tp6vup54) [inline components/script/dom/bindings/codegen/CodegenRust.py]: If braces are removed here, there will be an error occurred.

`error[E0429]: `self` imports are only allowed within a { } list`

Seems like the braces cannot be removed?

Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Ah, right, you need to remove `::{self}` entirely, i.e. the final declaration would look like `std::panic`.

**Comment by KiChjang** [comment]
Looks great! All that's left now is to squash your commits into one.

**Comment by KiChjang** [comment]
@bors-servo retry
* #13059

**Pull request merged**